image-analysis/api/image-analysis-swagger.yml

```diff
@@ -0,0 +1,188 @@
+################################################################################
+
+definitions:
+
+############
+## REQUESTS
+############
+  Image:
+    type: object
+    description: |
+      Data transfer object for an image.
+    properties:
+      ImageData:
+        type: string
+        description: The captured image data in PNG format, encoded as a base64 string. The data string shall not exceed 1MB.
+        example: iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAIAAACQd1PeAAAAEElEQVR4nGJiYGAABAAA//8ADAADcZGLFwAAAABJRU5ErkJggg==
+        x-nullable: false
+    required:
+      - ImageData
+
+#############
+## RESPONSES
+#############
+
+  Info:
+    type: object
+    description: Basic information describing the algorithm.
+    properties:
+      AlgorithmName:
+        type: string
+        description: Name of algorithm
+        example: AlwaysTrue
+      AlgorithmVersion:
+        type: string
+        description: Algorithm version identifier
+        example: "1.0.1"
+      AlgorithmType:
+        type: string
+        enum: [Face, Finger, Iris]
+        description: A string enum describing the type of biometric images the algorithm is meant to process
+      CompanyName:
+        type: string
+        description: Name of the Company which produces the algorithm
+        example: MdTF
+      TechnicalContactEmail:
+        type: string
+        description: The email address of an engineer or other technical resource to contact in the event of an error running your service. This field may be left blank if desired.
+        example: john@mdtf.org
+      RecommendedCPUs:
+        type: number
+        format: integer
+        description: The recommended allocation of CPUs for the deployed docker container.
+        example: 4
+      RecommendedMem:
+        type: number
+        format: integer
+        description: The recommended allocation of memory (MB) for the deployed docker container.
+        example: 2048
+    required:
+      - AlgorithmName
+      - AlgorithmVersion
+      - AlgorithmType
+      - CompanyName
+      - TechnicalContactEmail
+      - RecommendedCPUs
+      - RecommendedMem
+  
+  AnalysisScore:
+    type: number
+    format: float
+    minimum: 0
+    maximum: 1
+    description: Normalized score between 0 and 1, with 1 being the highest score the algorithm can produce, representing the result of analysis of the given biometric image.
+    example: 0.8734
+    x-nullable: false
+
+
+  RequestError:
+    type: string
+    description: |
+      Relevant and concise diagnostic information in the event of a client side error (e.g. malformed requests, or invalid image encoding).
+    example: Unable to decode image data as a PNG.
+
+  ServerError:
+    type: string
+    description: |
+      Relevant and concise diagnostic information in the event of a server side error (e.g. invalid license, or failure to generate a template).
+    example: The internal license has expired.
```

Comment:
lets change this error -- I dont want people running around with the idea they have to license these things -- maybe "array out of bounds" or something generic

Reply:
I'll use "Error encoding response" from the example implementation.